api/src/main/java/net/starype/quiz/api/game/ClassicalRound.java

```diff
@@ -0,0 +1,95 @@
+package net.starype.quiz.api.game;
+
+import net.starype.quiz.api.FixedLeaderboardEnding;
+import net.starype.quiz.api.game.LeaderboardDistribution.LeaderboardPosition;
+import net.starype.quiz.api.game.answer.Answer;
+import net.starype.quiz.api.game.event.EventHandler;
+import net.starype.quiz.api.game.player.UUIDHolder;
+import net.starype.quiz.api.game.question.Question;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+public class ClassicalRound implements GameRound {
+
+    private Question pickedQuestion;
+    private Collection<? extends UUIDHolder> players;
+    private MaxGuessCounter counter;
+    private double maxAwarded;
+    private LeaderboardDistribution leaderboard;
+    private QuizGame game;
+
+    public ClassicalRound(Question pickedQuestion, int maxGuesses, double maxAwarded) {
+        this.pickedQuestion = pickedQuestion;
+        this.counter = new MaxGuessCounter(maxGuesses);
+        this.maxAwarded = maxAwarded;
+    }
+
+    @Override
+    public void start(QuizGame game, Collection<? extends UUIDHolder> players, EventHandler eventHandler) {
+        this.game = game;
+        this.players = players;
+        this.leaderboard = new LeaderboardDistribution(maxAwarded, players.size());
+    }
+
+    @Override
+    public void onGuessReceived(UUIDHolder source, String message) {
+
+        Optional<Double> optCorrectness = pickedQuestion.evaluateAnswer(Answer.fromString(message));
+        if(optCorrectness.isEmpty()) {
+            return;
+        }
+
+        double correctness = optCorrectness.get();
+
+        counter.incrementGuess(source);
+        leaderboard.scoreUpdate(source, correctness);
+
+        if(Math.abs(correctness - 1) < 0.001) {
+            counter.consumeAllGuesses(source);
+        }
+
+        PlayerGuessContext context = new PlayerGuessContext(source, correctness, counter.isEligible(source));
+        game.sendInputToServer((server) -> server.onPlayerGuessed(context));
+    }
+
+    @Override
+    public void onGiveUpReceived(UUIDHolder source) {
+        counter.consumeAllGuesses(source);
+    }
+
+    @Override
+    public EntityEligibility initPlayerEligibility() {
+        return counter;
+    }
+
+    @Override
+    public RoundEndingPredicate initEndingCondition() {
+        return new NoGuessLeft(counter, players).or(new FixedLeaderboardEnding(leaderboard, players.size()));
+    }
+
+    @Override
+    public ScoreDistribution initScoreDistribution() {
+        return leaderboard;
+    }
+
+    @Override
+    public GameRoundReport initReport() {
+        return this::createReport;
+    }
+
+    private List<String> createReport() {
+        List<String> report = new ArrayList<>();
+        for(LeaderboardPosition position : leaderboard.getLeaderboard()) {
+            report.add(position.getPlayer().getUUID()+": " + position.getScore());

```

Comment:
I know that the ```ClassicalRound``` doesn't have any access to the Players class only to the ```UUIDHolder``` but i find it not really user friendly to print directly the UUID instead of the Players nickname or username.

Reply:
I was actually thinking that we should focus on the report system a little more since we haven't really worked on it seriously yet. For instance, here it would be nice if the report returned, as you said, the nicknames of the players, and also the score they obtained, not the accuracy. We might want to start working on another branch and pull request on this pull request

Reply:
Ok nice so we will let this is the current state for this PR